doozer/doozerlib/distgit.py

```diff
@@ -2720,6 +2725,68 @@ def rebase_dir(self, version: str, release: str, terminate_event, force_yum_upda
             self.rebase_event.set()  # awake all threads that are waiting for this image to be rebased
 
 
+class KonfluxImageDistGitRepo(ImageDistGitRepo):
+    """
+    It's not technically distgit anymore, but using the same name for simplicity
+    """
+    def __init__(self, metadata, autoclone=True):
+        super(KonfluxImageDistGitRepo, self).__init__(metadata, autoclone=False)
+
+        # Using k_distgits_dir which points to the new konflux dir
+        self.distgit_dir = os.path.join(self.runtime.k_distgits_dir, self.metadata.distgit_key)
+        self.dg_path = pathlib.Path(self.distgit_dir)
+
+        if autoclone:
+            self.clone()
+
+    def clone(self):
+        # Refresh if dir already exists, instead of cloning
+        if os.path.isdir(self.distgit_dir):
+            with Dir(self.distgit_dir):
+                self.logger.info(f"{self.distgit_dir} dir already exists, refreshing git instead of cloning")
+                exectools.cmd_assert('git fetch --all', retries=3)
+                exectools.cmd_assert('git reset --hard @{upstream}', retries=3)
+            return
+
+        branch = self.metadata.config.content.source.git.branch.target
+        url = self.metadata.config.content.source.git.url
+
+        git_args = ["--no-single-branch", "--branch", branch]
+        self.runtime.git_clone(url, self.distgit_dir, gitargs=git_args)
+
+    def push(self) -> None:
+        """
+        Push to the appropriate branch on openshift-priv
+        @return:
+        """
+        # Figure out which branch to push to
+        branch = f"art-<{self.runtime.group}>-assembly-<{self.runtime.assembly}>-dgk-<{self.name}>"
+        self.logger.info(f"Setting upstream branch to: {branch}")
+
+        with Dir(self.dg_path):
+            exectools.cmd_assert(f"git checkout -b {branch}")
+            exectools.cmd_assert(f"git push --set-upstream origin {branch}")
+
+    def add_distgits_diff(self, diff):
+        return self.runtime.add_distgits_diff(self.metadata.distgit_key, diff, konflux=True)
+
+    def update_distgit_dir(self, version, release, prev_release=None, force_yum_updates=False):
+        version, release = super().update_distgit_dir(version, release, prev_release, force_yum_updates)
+
+        # DNF repo injection steps for Konflux
+        dfp = DockerfileParser(path=str(self.dg_path.joinpath('Dockerfile')))
+        df_lines = dfp.content.splitlines(False)
+
+        df_lines.append("RUN mkdir -p /tmp/yum_temp; cp /etc/yum.repos.d/* /tmp/yum_temp/*; rm -rf /etc/yum.repos.d/*")
```

Comment:
Populating the repo file needs to happen after every `FROM` before the original Dockerfile can invoke yum/dnf. Each `FROM` indicates a new, independent stage in the build. 
Environment variable injection does this.

Reply:
added in https://github.com/openshift-eng/art-tools/pull/543/commits/20ff4db345bfc9f5275e44da5753ae9d7cec7e5c

new Dockerfile sample [here](https://github.com/ashwindasr/ocp-build-data/commit/cf4f9759ee0fad448867f11fdb83fefdb9850873#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557)